docs: document AVM v13 family box access (#639)
algorandfoundation/devportal: src/content/docs/concepts/smart-contracts/storage/box.mdx

Box storage in Algorand is a feature that provides additional on-chain storage options for smart contracts, allowing them to store and manage larger amounts of data beyond the limitations of global and local state. Unlike the fixed sizes of global and local state storages, box storage offers dynamic flexibility for creating, resizing, and deleting storage units

> These storage units, called boxes, are key-value storage segments associated with individual applications, each capable of storing upto 32KB (32768 bytes) of data as byte arrays. Boxes are only visible and accessible to the application that created them, ensuring data integrity and security. The app account (the smart contract) is responsible for funding the box storage, and only the creating app can read, write, or delete its boxes on-chain. Starting with AVM v13, an owning application can opt in to sharing its boxes with other applications; see [Sharing Boxes Between Applications](#sharing-boxes-between-applications).

Reviewer: ```suggestion
These storage units, called boxes, are key-value storage segments associated with individual applications, each capable of storing up to 32KB (32768 bytes) of data as byte arrays. The app account (the smart contract) is responsible for funding the box storage, and by default only the application that created a box can read, write, or delete it on-chain. Starting with AVM v13, an owning application can opt in to sharing its boxes with other applications; see [Sharing Boxes Between Applications](#sharing-boxes-between-applications).
```

Just tightening the paragraph and adding the "by default" qualifier for clarity.

(2500) + (400 * (1024+4)) = 413,700 microAlgos
```

> The minimum balance requirement always belongs to the application that owns the box, never to the application performing the operation. When a permitted application creates, resizes, or deletes a box owned by another application, it is the owning application's account that is affected. If that account cannot cover the increased requirement, the operation fails.

Reviewer: Might be worth mentioning here that with `AppSizeSponsor` the MBR for application extra pages and global state _can_ actually be covered by another account.

Author: `AppSizeSponsor` covers only extra program pages and the global state schema; this section is specifically about the box MBR requirement. My concern is that it creates an association while denying it—a skimming reader may see “`AppSizeSponsor` / box MBR / sponsorship” and take away the opposite intent.

Reviewer: Yeah I get what you mean. I think the "always" wording is what tipped me off. This is definitely a nit and shouldn't block the PR

### Reading

Boxes can be manipulated by the smart contract that owns them. The SDKs and goal cmd tool allow any box to be read off-chain, and box contents are never confidential; the restriction governs only which applications may read or manipulate a box on-chain.

> App a can read the contents of its boxes on-chain. Since AVM v13, app a can permit any application to read its boxes on-chain by setting the `AppForeignBoxReads` parameter; see [Foreign box reads](#foreign-box-reads). Recall that anybody can read everything from off-chain using the algod or indexer APIs.

Reviewer: ```suggestion
App a can read the contents of its boxes on-chain. Since AVM v13, app a can permit any application to read its boxes on-chain by setting the `AppForeignBoxReads` or `AppFamilyBoxAccess` parameter; see [Foreign box reads](#foreign-box-reads). Recall that anybody can read everything from off-chain using the algod or indexer APIs.
```

Other family members would need `AppFamilyBoxAccess`, not `AppForeignBoxReads`
